core/src/main/java/de/danoeh/antennapod/core/service/playback/PlaybackService.java

```diff
@@ -552,68 +561,32 @@ private void displayStreamingNotAllowedNotification(Intent originalIntent) {
      * Handles media button events
      * return: keycode was handled
      */
-    private boolean handleKeycode(int keycode, boolean notificationButton) {
+    private boolean handleKeycode(int keycode, boolean mediaButton) {
         Log.d(TAG, "Handling keycode: " + keycode);
         final PlaybackServiceMediaPlayer.PSMPInfo info = mediaPlayer.getPSMPInfo();
         final PlayerStatus status = info.playerStatus;
         switch (keycode) {
             case KeyEvent.KEYCODE_HEADSETHOOK:
             case KeyEvent.KEYCODE_MEDIA_PLAY_PAUSE:
-                if (status == PlayerStatus.PLAYING) {
-                    mediaPlayer.pause(!UserPreferences.isPersistNotify(), false);
-                } else if (status == PlayerStatus.PAUSED || status == PlayerStatus.PREPARED) {
-                    mediaPlayer.resume();
-                } else if (status == PlayerStatus.PREPARING) {
-                    mediaPlayer.setStartWhenPrepared(!mediaPlayer.isStartWhenPrepared());
-                } else if (status == PlayerStatus.INITIALIZED) {
-                    mediaPlayer.setStartWhenPrepared(true);
-                    mediaPlayer.prepare();
-                } else if (mediaPlayer.getPlayable() == null) {
-                    startPlayingFromPreferences();
-                }
+                handleKeyAction(KeyEvent.KEYCODE_MEDIA_PLAY_PAUSE, mediaButton, MediaAction.TOGGLE_PLAYBACK);
```

Comment:
A decision I made here was to treat KEYCODE_HEADSETHOOK, KEYCODE_MEDIA_PLAY_PAUSE, KEYCODE_MEDIA_PLAY, and KEYCODE_MEDIA_PAUSE all as KEYCODE_MEDIA_PLAY_PAUSE when determining if a media event was 'multi-tapped'. So if a user sends a KEYCODE_HEADSETHOOK followed very quickly by a KEYCODE_MEDIA_PAUSE, it would use the play/pause double-tap behavior if the user had one configured.

For KEYCODE_HEADSETHOOK and KEYCODE_MEDIA_PLAY_PAUSE I think this okay, as we already handle them in the same way to 'toggle playback'.

For KEYCODE_MEDIA_PLAY and KEYCODE_MEDIA_PAUSE, I figured that a hardware device might implement their play/pause functionality by only sending PLAY or PAUSE based on their own internal playback state. Or, the hardware device might only have separate play and pause buttons and a user might still want to use the multi-tap features by hitting the two buttons in quick succession. I have to admit this part has not been tested much and am not sure if there is a huge use case for it.